docs/DesignDocs/SGX_QuoteEx_Integration.md

```diff
@@ -0,0 +1,127 @@
+- `sgx_get_quote_size_ex(const sgx_att_key_id_t *p_att_key_id, uint32_t* p_quote_size)`
+    - Return the size of the buffer needed to hold the quote to be generated for the given attestation key ID.
+- `sgx_get_quote_ex(const sgx_report_t *p_app_report, const sgx_att_key_id_t *p_att_key_id,sgx_qe_report_info_t *p_qe_report_info, uint8_t *p_quote, uint32_t quote_size)`
+    - Generate a quote for the given attestation key ID and application SGX report, and return it in the call-supplied buffer.
+
+As compared to the DCAP library API, the quote-ex library API allows enumeration of supported evidence formats (called attestation key IDs in the API). Otherwise the quote-ex API is similar to the DCAP API, except that every function takes an input attestation key ID in its parameter list.
+
+### Host-side Plugin Library Dynamic Load of the quote-ex or DCAP Library
+
+Between the DCAP and quote-ex libraries:
+- The DCAP library only supports generation of SGX quote in ECDSA-p256 format. With DCAP, the quote generation can be done either in-process, or out-of-process by working with a background service running on the same platform.
+- The quote-ex library supports generation of SGX quote in multiple formats (including ECDSA-p256 and EPID variations). With quote-ex, quote generation is always done out-of-process by working with a background service (called AESM) on the local platform.
+
+An SGX platform can have either the quote-ex library or the DCAP library, or both of them installed. When both libraries are installed, the quote-ex library takes precedence, as it supports more evidence formats.
+
+The OE SDK host-side plugin library is changed to dynamically detect the presence of the two libraries, load one of them, and record the entry points of the loaded library for invocation in the flows described below.
+
+#### Alternative: Only Use the quote-ex Library
+
+Run-time dynamic detection and loading of multiple shared libraries complicates implementation. It also increases risk of version mismatch between OE SDK and the dependent libraries that is not detected at build time via library headers.
+
+As described previously, the quote-ex library supports a superset of formats as compared to the DCAP library, though it always depends on a background service for quote generation. If on SGX platforms the OE SDK always installs with the AESM background service (as a hard dependency), then an alternative implementation is to use the libsgx-quote-ex library only (instead of the DCAP library). With this implementation, the libsgx-quote-ex library is linked to the OE SDK at build time.
+
+### Support of SGX Evidence Formats Enumeration
+
+The SGX plugin code file `enclave/sgx/attester.c` implements the OE SDK API `oe_sgx_get_attesters()`. The implementation enumerates all supported SGX evidence formats, creates a list of attester plugins for them, and returns the created list to the caller.
```

Comment:
I believe oe_sgx_get_attesters() should be oe_get_attester_plugins() like you have in line 13

Reply:
Yes, thank you!